gui/elements/_viewer.py

```diff
@@ -239,3 +242,46 @@ def _on_layers_change(self, event):
         """
         self._child_layer_changed = True
         self._update()
+
+    def on_mouse_move(self, event):
+        """Called whenever mouse moves over canvas.
+        """
+        if event.pos is None:
+            pos = None
+        else:
+            visual = self.layers[0]._node
+            tr = self._canvas.scene.node_transform(self.layers[0]._node)
+            pos = tr.map(event.pos)
+            self._pos = [clip(pos[0],0,self.max_shape[0]-1), clip(pos[1],0,self.max_shape[1]-1)]
+            self.update_statusBar()
+
+    def update_statusBar(self):
+        from ..layers._image_layer import Image
+
+        msg = 'x=%d, y=%d' % (self._pos[0], self._pos[1])
+        if self.max_dims > 2:
+            for i in range(2,self.max_dims):
+                msg = msg + ', %d=%d' % (i, self.indices[i])
+        top = None
+        for i in range(len(self.layers)):
+            if self.layers[i].visible and isinstance(self.layers[i], Image):
+                top = i
+        if top is None:
+            top = None
+        else:
+            if self.max_dims > 2:
+                ind = int(self._pos[1]),int(self._pos[0]),*self.indices[2:]
+                value = self.layers[top].image[ind]
```

Comment:
This may throw an `IndexError`.

Reply:
not sure I understand. Is the problem the use of `x`? Let me know any alternatives and I'll change it.

Reply:
This will be out of bounds whenever the `viewer.max_shape` is greater than the `layer.shape` or the cursor position is out of bounds of this shape. I would recommend implementing this in a similar fashion to `layer._set_viewer_slice`.

Reply:
Ah yes - gets at my question of what are we doing about the cases where not all of the images have the same dimensions - will checkout `layer._set_viewer_slice`

Reply:
I have now done this and pushed the changes, let me know what you think